File: Mak_V3/Chapter18/src/HelloMany.java
public class HelloMany
{
    private static RunTimer     _runTimer;
    private static PascalTextIn _standardIn;

    private static int count;

    private static void sayHello(int howManyTimes)
    {
        int i;

        for (i = 1; i <= howManyTimes; ++i) {
            System.out.print(String.format("%3d: Hello, world.\n", i));
        }
    }

    public static void main(String[] args)
        throws Exception
    {
        _runTimer = new RunTimer();
        _standardIn = new PascalTextIn();

        System.out.print("How many times? ");
        count = _standardIn.readInteger();
        System.out.println();

        sayHello(count);

        _runTimer.printElapsedTime();
    }
}

File: Mak_Writing_Compiler_3rd_Ed/Chapter18/HelloMany.pas
PROGRAM HelloMany;

VAR
    count : integer;

PROCEDURE sayHello(howManyTimes : integer);

    VAR
        i : integer;

    BEGIN
        FOR i := 1 TO howManyTimes DO BEGIN
            writeln(i:3, ': ', 'Hello, world.')
        END
    END;

BEGIN {HelloMany}
    write('How many times? ');
    read(count);
    writeln;

    sayHello(count);
END.
